build_projects/shared-build-targets-utils/Publishing/AzurePublisher.cs

```diff
@@ -134,14 +146,62 @@ public IEnumerable<string> ListBlobs(string virtualDirectory)
                 }
             }
 
+            ResetLeaseRenewalTaskState();
```

Comment:
We are about throw an exception and thus, do not expect any lease to be acquired. Hence, we reset the Task.